crypto/rsa/rsa_oaep.c

```diff
@@ -208,7 +208,7 @@ int RSA_padding_check_PKCS1_OAEP_mgf1(unsigned char *to, int tlen,
 
     mdlen = EVP_MD_get_size(md);
 
-    if (tlen <= 0 || flen <= 0)
+    if (tlen <= 0 || flen <= 0 || mdlen < 0)
```

Comment:
Please use `mdlen <= 0` as there is no point in digest with zero length and we have the same checks elsewhere.